lower.ts

```diff
@@ -113,6 +113,27 @@ function flattenStmt(s : AST.Stmt<Type>, blocks: Array<IR.BasicBlock<Type>>, env
     // ]];
   
     case "expr":
+      // list-comp changes
+      if(s.tag == "expr" && s.expr.tag == "list-comp")
```

Comment:
Doesn't the case already check that the tag is "expr"?

Reply:
This has been fixed.